packages/massa-web3/src/web3/SmartContractsClient.ts

```diff
@@ -405,6 +426,52 @@ export class SmartContractsClient
     )
   }
 
+  /**
+   * Watches the status of an operation and invokes a callback function when the status changes.
+   *
+   * @param opId - The ID of the operation to watch.
+   * @param callback - The function to call when the operation status changes. It receives the new status as a parameter.
+   * @param timeInterval - The interval in milliseconds at which to check the operation status. Defaults to `TX_POLL_INTERVAL_MS`.
+   * @param timeout - The time at which to stop checking the operation status. Defaults to the current time plus `WAIT_OPERATION_TIMEOUT`.
+   *
+   * @returns A Promise that resolves to a function that can be called to stop watching the operation status.
+   */
+  public async watchOperationStatus(
+    opId: string,
+    callback: (status: EOperationStatus, error?: Error) => void,
+    timeInterval = TX_POLL_INTERVAL_MS,
+    timeout = Date.now() + WAIT_OPERATION_TIMEOUT
```

Comment:
its strange to have `Date.now()` here. usually we set a timeout range (like 3 secondes) implicitely starting from now.

Reply:
I updated the code, I don't know if it's what you was saying